### PR TITLE
Add support for applying JSON string

### PR DESCRIPTION
**Pull request by yahu577**
Currently there's no way to supply array based json configuration. This addition allow you to apply the env configuration as a any json object, which also allows supplying Arrays.

### CODE REVIEW

File: lib/nconf/stores/env.js [hunk 1]
```diff
@@ -15,7 +15,7 @@ var util = require('util'),
 // Constructor function for the Env nconf store, a simple abstraction
 // around the Memory store that can read process environment variables.
 //
-var Env = exports.Env = function (options) {
+var Env = exports.Env = function(options) {
```
Comment (indexzero) [inline lib/nconf/stores/env.js]: Why change this?

File: lib/nconf/stores/env.js [hunk 2]
```diff
@@ -68,11 +68,27 @@ Env.prototype.loadEnv = function () {
       return !self.whitelist.length || self.whitelist.indexOf(key) !== -1
     }
   }).forEach(function (key) {
+    var _val = (function(val) {
```
Comment (indexzero) [inline lib/nconf/stores/env.js]: Why use an immediately invoked function expression here?

**Comment by indexzero** [comment]
See comments above. Why use `JSON` and not something like a `,` separated value?

**Comment by yahu577** [comment]
@indexzero 
1. Fixed comments. 
2. Indeed, allowing JSON support is more robust function to me to override any configuration attribute type from Env provider, as currently you can only override string value types with Env provider. I would even suggest to support Number or Boolean as well, however, I don't want to break compatibility. What's your thoughts on this?

**Comment by indexzero** [comment]
@yahu577 awesome. Provide some tests and this will make it in.

**Comment by AdrieanKhisbe** [comment]
Not sure it's a good this that this is applied by default.
Might break compatibility, and it think it's better that it happens only on demand. :)

**Comment by xizhao** [comment]
Is this still happening?

**Comment by mhamann** [comment]
Merged via #272

**Pull request closed without merging**